addons/mail/static/src/discuss/discuss.js

```diff
@@ -27,6 +49,34 @@ export class Discuss extends Component {
     startCall() {
         this.messaging.startCall(this.messaging.discuss.threadId);
     }
+
+    /**
+     * @param {KeyboardEvent} ev
+     */
+    onKeyDownThreadNameInput(ev) {
+        switch (ev.key) {
+            case "Enter":
+                this.renameThread();
+                this.threadNameInputRef.el.blur();
+                break;
+            case "Escape":
+                this.state.threadRename = this.currentThread().name;
+                this.threadNameInputRef.el.blur();
+                break;
+        }
+    }
+
+    async renameThread() {
+        const newName = this.state.threadRename.trim();
+        const thread = this.currentThread();
+        if (
+            (newName && newName !== thread.name && thread.type === "channel") ||
+            (newName !== thread.name &&
```

Comment:
`newName !== thread.name` in different disjunctive branches, when this is always a necessary condition for `notifyThreadNameToServer`.

```js
newName !== thread.name &&
(newName && thread.type === "channel" || thread.type === "chat" || thread.type === "group")
```